plutus-core/plutus-core/src/PlutusCore/Parser/Type.hs

```diff
@@ -46,9 +46,15 @@ ifixType = withSpan $ \sp ->
 
 builtinType :: Parser PType
 builtinType = withSpan $ \sp -> inParens $ do
-    SomeTypeIn (Kinded uni) <- (symbol "con" *> defaultUni)
+    SomeTypeIn (Kinded uni) <- symbol "con" *> defaultUni
     pure $ TyBuiltin sp (SomeTypeIn uni)
 
+sopType :: Parser PType
+sopType = withSpan $ \sp -> inParens $ TySOP sp <$> (symbol "sop" *> many tyList)
+  where
+    tyList :: Parser [PType]
+    tyList = (inBrackets $ many pType) <* whitespace
```

Comment:
I think this is @zliu41 . The whitespace handling became complicated once we started tracking source spans (this could do with a note TBH). Now we have to be careful about consuming trailing whitespace otherwise it will make the end of the source span look wrong. The correct solution would be a lexer, I think...

Reply:
> Now we have to be careful about consuming trailing whitespace otherwise it will make the end of the source span look wrong.

Ah, of course, thanks.